pkg/policies/policies.go

```diff
@@ -16,13 +16,179 @@
 package policies
 
 import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+
 	v1 "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/internal/attestation/crafter"
+	v12 "github.com/chainloop-dev/chainloop/internal/attestation/crafter/api/attestation/v1"
+	"github.com/chainloop-dev/chainloop/internal/casclient"
 	"github.com/chainloop-dev/chainloop/pkg/policies/engine"
 	"github.com/chainloop-dev/chainloop/pkg/policies/engine/rego"
+	"github.com/sigstore/cosign/v2/pkg/blob"
+	"google.golang.org/protobuf/encoding/protojson"
 )
 
-// GetPolicyEngine returns a PolicyEngine implementation to evaluate a given policy.
-func GetPolicyEngine(_ *v1.Policy) engine.PolicyEngine {
+type PolicyVerifier struct {
+	state *v12.CraftingState
+	cas   casclient.Downloader
+}
+
+func NewPolicyVerifier(state *v12.CraftingState, client casclient.Downloader) *PolicyVerifier {
+	// only Rego engine is currently supported
+	return &PolicyVerifier{state: state, cas: client}
+}
+
+// Verify verifies that the statement is compliant with the policies present in the schema
+func (pv *PolicyVerifier) Verify(ctx context.Context) ([]*engine.PolicyViolation, error) {
+	violations := make([]*engine.PolicyViolation, 0)
+	policies := pv.state.GetInputSchema().GetPolicies()
+	for _, policyAtt := range policies {
+		if policyAtt.Disabled {
+			// policy is disabled
+			// TODO: WARN.
```

Comment:
Shall we add the warn?

Reply:
In the next PR, I'm injecting the logger. I'll do it there. Thanks for the heads up.